tools/flecstan/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# -*- CMake -*-
+
+project(flecstan LANGUAGES CXX)
+cmake_minimum_required(VERSION 3.8)
+
+# Use cmake modules from cinch
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../cinch/cmake)
+
+# C++
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -O2 -s")
+
+# Packages
+find_package(Clang REQUIRED COMPONENTS Tooling)
+
+# Include directories
+include_directories(src ${CLANG_INCLUDE_DIRS} ${LLVM_INCLUDE_DIRS})
+
+# Source files
+set(flecstan_SOURCES
+   src/flecstan-analysis.cc
+   src/flecstan-arg.cc
+   src/flecstan-diagnostic.cc
+   src/flecstan-macro.cc
+   src/flecstan-main.cc
+   src/flecstan-misc.cc
+   src/flecstan-prep.cc
+   src/flecstan-utility.cc
+   src/flecstan-visitor.cc
+   src/flecstan-yaml.cc
+)
+
+# Executable
+add_executable(flecstan ${flecstan_SOURCES})
+
+# Libraries
+target_link_libraries(
+   flecstan clangTooling
```

Comment:
Drop `clangTooling` this is part of `${CLANG_LIBRARIES}` already.

Reply:
Will try. It should be part of the clang libraries already, but this was one element I believe was sometimes not being found, so it seemed reasonable to give it some extra encouragement. :-)

Reply:
Right now, even on just my own machine (much less everything in the CI), the build has link errors if I don't *explicitly* say clangTooling in addition to ${CLANG_LIBRARIES}. This has been a problem all along. Perhaps CMake has a flaw with regards to its understanding of the Clang libraries.

Reply:
Hmm that is kind of strange, isn't `find_package(Clang REQUIRED COMPONENTS Tooling)` exactly trying to find `libClangTooling` ?

Reply:
I just check on my machine (Fedora 29), `CLANG_LIBRARIES` contains `/usr/lib64/libclangTooling.so`, which is what I expected.

Reply:
.so rather than .a. Hmm. I wonder if there could be an issue with static versus dynamic linking in all these cmake follies.

Reply:
Yes, one would think that find_package(Clang REQUIRED COMPONENTS Tooling) would be finding libclangTooling .a or .so, but its apparent failure to do so seems to be the crux of this build issue.